src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +122,53 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter)
+                    && (previouslyMatchedEpisodes == other.previouslyMatchedEpisodes);
         }
     };
 }
 
 using namespace RSS;
 
+static const QString episodeRegex = "(?:^|[^A-Za-z0-9])((?:S\\d{1,}(?:E\\d{1,})+)|(?:\\d{1,}x\\d{1,})|(?:\\d{4}\\.\\d{2}\\.\\d{2}))(?:[^A-Za-z0-9]|$)";
+
+QString computeEpisodeName(const QString &article)
+{
+    QRegExp f(episodeRegex, Qt::CaseInsensitive);
+    // See if we can extract an season/episode number or date from the title
+    if (f.indexIn(article) == -1) {
+        return QString();
+    }
+
+    // Extract out the episode identifier
+    QString matchedEpisode = f.cap(1);
+
+    // Is it in the S01E01 format?
+    if (matchedEpisode.startsWith('S', Qt::CaseInsensitive)) {
```

Comment:
I just tried doing this with named groups, but it seems that duplicate named groups are not allowed. This means I'd need to use 5 separate groups and a bunch of if/else if to extract the correct one. I remember having the issue with normal groups, hence how the code is written at the moment.

I'll see if I can rewrite that method to be simpler.